docs/DeveloperGuide.adoc

```diff
@@ -297,57 +297,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |user |untrack a module |remove entries that I no longer need
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |user |find a module |locate details of modules that NUS provides
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |curious student |view a module's details |check out if I want to take the module
+
+|`* * *` |busy student |manage deadlines |keep track of my assignments and tasks
+
+|`* * *` |organized student |manage my grades |know roughly how well I'm doing in the module
+
+|`* *` |new student |schedule my modules |plan how to approach my future focus areas
+
+|`* *` |graduating student |check what modules I need to complete |
+
+|`* *` |new student |schedule my modules |plan how to approach my future focus areas
+
+|`* *` |computing student |plan my focus area modules |choose a specialization
+
+|`* *` |competitive student |calculate my GPA |
+
+|`* *` |busy student |prioritize certain modules (and their deadlines) |manage my time better
+
+|`* *` |organized student |manage files related to a module |add and tag files
+
+|`* *` |student |catalog past modules |access the content conveniently
+
+|`*` |student |generate a sample module planner |plan my modules
+
+|`*` |hardworking student |access past exam papers |
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `ModuleBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Find and Track a Module
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to find a module with incomplete keyword (e.g. 'CS2')
+2.  ModuleBook shows a list of modules matching the keyword (e.g. 'CS2030, CS2040, CS2100, CS2101, etc.')
+3.  User requests to track a module
+4.  ModuleBook tracks the module
 +
 Use case ends.
 
-*Extensions*
+[discrete]
+=== Use case: Add a deadline for a tracked module
 
-[none]
-* 2a. The list is empty.
+*MSS*
+
+1.  User requests to view the specific tracked module
+2.  ModuleBook displays the module details
+3.  User requests to add a deadline
+4.  ModuleBook adds the deadline
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
+
+[none]
+* 2a. The module is not currently tracked
+* 2b. User requests to track the module
+* 2c. The module is now tracked
 +
+Use case resumes at step 3
```

Comment:
This part is confusing, perhaps you could insert  step 3a onwards after step 3 as step 4 and so on. Step 3d1 and 3d2 should be extensions.

Reply:
Removed for now. We can probably amend this next tme.